src/Features/LanguageServer/ProtocolUnitTests/Completion/CompletionTests.cs

```diff
@@ -257,6 +306,35 @@ void M()
             Assert.True(testAccessor.GetCacheContents().Count == 3);
         }
 
+        [Fact]
+        public async Task TestGetCompletionsWithDeletionInvokeKindAsync()
+        {
+            var markup =
+@"class A
+{
+    void M()
+    {
+        {|caret:|}
+    }
+}";
+            using var workspace = CreateTestWorkspace(markup, out var locations);
+            var completionParams = CreateCompletionParams(
+                locations["caret"].Single(),
+                invokeKind: LSP.VSCompletionInvokeKind.Deletion,
+                triggerCharacter: "M",
+                triggerKind: LSP.CompletionTriggerKind.Invoked);
+
+            var document = workspace.CurrentSolution.Projects.First().Documents.First();
+
+            var expected = await CreateCompletionItemAsync("A", LSP.CompletionItemKind.Class, new string[] { "Class", "Internal" },
+                completionParams, document, commitCharacters: CompletionRules.Default.DefaultCommitCharacters).ConfigureAwait(false);
+
+            var results = await RunGetCompletionsAsync(workspace.CurrentSolution, completionParams).ConfigureAwait(false);
+
+            // By default, completion doesn't trigger on deletion.
+            Assert.Null(results);
```

Comment:
aside, @allisonchou @dibarbet we really need to thinka bout how we're going to get all of our existing completion tests running under LSP.  First, it will help us find any more holes in the support between the two systems.  Second, it will actually show how those holes are being plugged as it will allow us to unskip tests that dont' work on LSP.  Third, it will give us very good coverage of the LSP completion stack.  

It's really not good that we're testing things this way.  It's basically doesn't give me confidence that LSP is providing the right behavior taht we expect of deletion/trigger-chars.

Reply:
Agreed, would def be ideal to have the existing completion tests running in the LSP world as well.